File: AssetManager.Api/Models/Asset3D.cs
using AssetManager.Api.Models.Enums;
namespace AssetManager.Api.Models;

public class Asset3D
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Category { get; set; }
    public string? Tags { get; set; }

    public string OriginalFileName { get; set; } = string.Empty;
    public string StoredFileName { get; set; } = string.Empty;
    public string FilePath { get; set; } = string.Empty;
    public string? ThumbnailPath { get; set; }

    public long FileSize { get; set; }

    public int? VertexCount { get; set; }
    public int? TriangleCount { get; set; }
    public int? MeshCount { get; set; }
    public int? MaterialCount { get; set; }
    public int? AnimationCount { get; set; }

    public decimal? Width { get; set; }
    public decimal? Height { get; set; }
    public decimal? Depth { get; set; }

    public AssetStatus Status { get; set; } = AssetStatus.Processing;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public List<ExportedAssetFile> ExportedFiles { get; set; } = new();
}

File: asset-manager-api-node/src/domain/entities/Asset3D.ts
import { AssetStatus } from "../enums/AssetStatus";
import { ExportedAssetFile } from "./ExportedAssetFile";

export class Asset3D {
  constructor(
    public readonly id: string,
    public name: string,
    public description: string | null,
    public category: string | null,
    public tags: string | null,

    public originalFileName: string,
    public storedFileName: string,
    public filePath: string,
    public thumbnailPath: string | null,

    public fileSize: number,

    public vertexCount: number | null,
    public triangleCount: number | null,
    public meshCount: number | null,
    public materialCount: number | null,
    public animationCount: number | null,

    public width: number | null,
    public height: number | null,
    public depth: number | null,

    public status: AssetStatus,

    public readonly createdAt: Date,
    public updatedAt: Date,

    public exportedFiles: ExportedAssetFile[] = []
  ) {}
}
